src/client/clientenvironment.cpp

```diff
@@ -361,26 +361,27 @@ GenericCAO* ClientEnvironment::getGenericCAO(u16 id)
 	return NULL;
 }
 
-u16 ClientEnvironment::addActiveObject(ClientActiveObject *object)
+u16 ClientEnvironment::addActiveObject(std::unique_ptr<ClientActiveObject> object)
 {
+	auto obj = object.get();
 	// Register object. If failed return zero id
-	if (!m_ao_manager.registerObject(object))
+	if (!m_ao_manager.registerObject(std::move(object)))
```

Comment:
Using `obj` after moving `object` doesn’t feel safe. I know it is here and now, but... Murphy awaits.
And I see this pattern all over the ActiveObject commit.

Reply:
The semantics are the same as before. So this feeling of unsafety is a good thing, it makes one more wary.

The obj should still be alive if the returned id is not 0, I guess. This is checked everywhere (apart from some places in the unittests).